libs/core/include/core/serializers/stl_types.hpp

```diff
@@ -17,318 +17,4 @@
-
-  // Reset the map
-  map.clear();
-
-  // Update the map
-  K key{};
-  V value{};
-  for (uint64_t i = 0; i < size; ++i)
-  {
-    serializer >> key;
-    serializer >> value;
-
-    map[key] = value;
-  }
-}
-
-template <typename T, typename K, typename V>
-inline void Serialize(T &serializer, std::map<K, V> const &map)
-{
-  // Allocating memory for the size
-  serializer.Allocate(sizeof(uint64_t));
-
-  uint64_t size = map.size();
-
-  // Writing the size to the byte array
-  serializer.WriteBytes(reinterpret_cast<uint8_t const *>(&size), sizeof(uint64_t));
-
-  for (auto const &element : map)
-  {
-    serializer << element.first << element.second;
-  }
-}
-
-template <typename T, typename K, typename V>
-inline void Deserialize(T &serializer, std::map<K, V> &map)
-{
-
-  // Read the number of items in the map
-  uint64_t size{0};
-  serializer.ReadBytes(reinterpret_cast<uint8_t *>(&size), sizeof(uint64_t));
-
-  // Reset the map
-  map.clear();
-
-  // Update the map
-  K key{};
-  V value{};
-  for (uint64_t i = 0; i < size; ++i)
-  {
-    serializer >> key;
-    serializer >> value;
-
-    map[key] = value;
-  }
-}
-
-template <typename T, typename K, typename H>
-inline void Serialize(T &serializer, std::unordered_set<K, H> const &set)
-{
-
-  // Allocating memory for the size
-  serializer.Allocate(sizeof(uint64_t));
-
-  uint64_t size = set.size();
-
-  // Writing the size to the byte array
-  serializer.WriteBytes(reinterpret_cast<uint8_t const *>(&size), sizeof(uint64_t));
-
-  for (auto const &element : set)
-  {
-    serializer << element;
-  }
-}
-
-template <typename T, typename K, typename H>
-inline void Deserialize(T &serializer, std::unordered_set<K, H> &set)
-{
-
-  // Read the number of items in the map
-  uint64_t size{0};
-  serializer.ReadBytes(reinterpret_cast<uint8_t *>(&size), sizeof(uint64_t));
-
-  // Reset the map
-  set.clear();
-
-  // Update the map
-  K key{};
-  for (uint64_t i = 0; i < size; ++i)
-  {
-    serializer >> key;
-    set.insert(key);
-  }
-}
-
-template <typename T, typename K>
-inline void Serialize(T &serializer, std::set<K> const &set)
-{
-
-  // Allocating memory for the size
-  serializer.Allocate(sizeof(uint64_t));
-
-  uint64_t size = set.size();
-
-  // Writing the size to the byte array
-  serializer.WriteBytes(reinterpret_cast<uint8_t const *>(&size), sizeof(uint64_t));
-
-  for (auto const &element : set)
-  {
-    serializer << element;
-  }
-}
-
-template <typename T, typename K>
-inline void Deserialize(T &serializer, std::set<K> &set)
-{
-
-  // Read the number of items in the map
-  uint64_t size{0};
-  serializer.ReadBytes(reinterpret_cast<uint8_t *>(&size), sizeof(uint64_t));
-
-  // Reset the map
-  set.clear();
-
-  // Update the map
-  K key{};
-  for (uint64_t i = 0; i < size; ++i)
-  {
-    serializer >> key;
-    set.insert(key);
-  }
-}
-
-}  // namespace serializers
-}  // namespace fetch
+#include "core/serializers/base_types.hpp"
```

Comment:
Can we make an issue to remove this file and simply include the new one (maybe worth doing once files have moved into final place

Reply:
It's removed.